apis/v1alpha2/policy_types.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+// PolicyTargetReference identifies an API object to apply policy to. This
+// should be used as part of Policy resources that can target Gateway API
+// resources. For more information on how this policy attachment model works,
+// and a sample Policy resource, refer to the policy attachment documentation
+// for Gateway API.
+type PolicyTargetReference struct {
+	// Group is the group of the target resource.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Group string `json:"group"`
+
+	// Kind is kind of the target resource.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Kind string `json:"kind"`
+
+	// Name is the name of the target resource.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name"`
```

Comment:
Question: how do I target a policy to an entire namespace if this is required?

Reply:
Need to add an example of this at some point, but my thought is that it would be like any other reference:

```yaml
- group: ""
  kind: "Namespace"
  name: "foo"
```

Reply:
> Need to add an example of this at some point, but my thought is that it would be like any other reference:
> 
> ```yaml
> - group: ""
>   kind: "Namespace"
>   name: "foo"
> ```

Interesting.  If you mean "target all resources within a namespace", I woulda thunk I'd need this:

```yaml
- group: ""
  kind: ""
  name: ""
  namespace: "foo"
```

Reply:
Anyway, doesn't at least `Group` need to have `MinLength=0`, or be an optional pointer value?